pkg/db/db_test.go

```diff
@@ -1823,6 +1837,1151 @@ func TestReadReleasesByApp(t *testing.T) {
+											DeployAuthor: "testmail@example.com",
+											DeployTime:   "1",
+										},
+										Team: "team-123",
+									},
+								},
+								Locks: map[string]*api.Lock{
+									"dev-lock": {
+										LockId:    "dev-lock",
+										Message:   "My lock on dev",
+										CreatedAt: timestamppb.New(time.Date(2024, time.July, 12, 15, 30, 0, 0, time.UTC)),
+										CreatedBy: &api.Actor{
+											Name:  "myself",
+											Email: "myself@example.com",
+										},
+									},
+								},
+								Priority: api.Priority_YOLO,
+							},
+						},
+						Priority: api.Priority_YOLO,
+					},
+				},
+				Applications: map[string]*api.Application{
+					"test": {
+						Name: "test",
+						Releases: []*api.Release{
+							{
+								Version:        1,
+								SourceCommitId: "deadbeef",
+								SourceAuthor:   "example <example@example.com>",
+								SourceMessage:  "changed something (#678)",
+								PrNumber:       "678",
+								CreatedAt:      &timestamppb.Timestamp{Seconds: 1, Nanos: 1},
+							},
+						},
+						Team: "team-123",
+					},
+				},
+				GitRevision: "0",
+			},
+		},
+		{
+			Name: "env does not exists",
+			NewEnvironmentLock: EnvironmentLock{
+				Env:        "does-not-exists",
+				LockID:     "dev-lock",
+				EslVersion: 2,
+				Deleted:    false,
+				Metadata: LockMetadata{
+					Message:        "My lock on dev for my-team",
+					CreatedByName:  "myself",
+					CreatedByEmail: "myself@example.com",
+				},
+				Created: time.Date(2024, time.July, 12, 15, 30, 0, 0, time.UTC),
+			},
+			ExpectedError: errMatcher{"could not find environment does-not-exists in overview"},
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+			ctx := testutil.MakeTestContext()
+			dbHandler := setupDB(t)
+
+			err := dbHandler.WithTransaction(ctx, false, func(ctx context.Context, transaction *sql.Tx) error {
+				err := dbHandler.WriteOverviewCache(ctx, transaction, startingOverview)
+				if err != nil {
+					return err
+				}
+				err = dbHandler.UpdateOverviewEnvironmentLock(ctx, transaction, tc.NewEnvironmentLock)
+				if err != nil {
+					if diff := cmp.Diff(tc.ExpectedError, err, cmpopts.EquateErrors()); diff != "" {
+						return fmt.Errorf("mismatch between errors (-want +got):\n%s", diff)
+					}
+					return nil
+				}
+				latestOverview, err := dbHandler.ReadLatestOverviewCache(ctx, transaction)
+				if err != nil {
+					return err
+				}
+				opts := cmpopts.IgnoreUnexported(api.GetOverviewResponse{}, api.EnvironmentGroup{}, api.Environment{}, api.Application{}, api.Release{}, timestamppb.Timestamp{}, api.EnvironmentConfig{}, api.EnvironmentConfig_Upstream{}, api.Environment_Application{}, api.Environment_Application_DeploymentMetaData{}, api.EnvironmentConfig_ArgoCD{}, api.Lock{}, api.Actor{})
+				if diff := cmp.Diff(tc.ExcpectedOverview, latestOverview, opts); diff != "" {
```

Comment:
maybe extract this into a function like `getOverviewIgnoredTypes`, as it's used at least twice.